packages/loader/driver-utils/src/adapters/compression/summaryblob/documentStorageServiceSummaryBlobCompressionAdapter.ts

```diff
@@ -0,0 +1,445 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IsoBuffer, assert } from "@fluidframework/common-utils";
+import { IDocumentStorageService, ISummaryContext } from "@fluidframework/driver-definitions";
+import {
+	ISnapshotTree,
+	ISummaryBlob,
+	ISummaryHandle,
+	ISummaryTree,
+	IVersion,
+	SummaryObject,
+	SummaryType,
+} from "@fluidframework/protocol-definitions";
+import { compress, decompress } from "lz4js";
+import { DocumentStorageServiceProxy } from "../../../documentStorageServiceProxy";
+import { ICompressionStorageConfig, SummaryCompressionAlgorithm } from "../";
+
+/**
+ * This class is a proxy for the IDocumentStorageService that compresses and decompresses blobs in the summary.
+ * The identification of the compressed blobs is done by adding a compression markup blob to the summary.
+ * Even if the markup blob is present, it does not mean that all blobs are compressed. The blob,
+ * which is compressed also contain the compression algorithm enumerated value from the
+ * SummaryCompressionAlgorithm enumeration in the first byte . If the blob is not
+ * commpressed, it contains the first byte equals to SummaryCompressionAlgorithm.None .
+ * In case, the markup blob is present, it is expected that the first byte of the markup blob
+ * will contain the info about the compression. If the first byte is not present, it is assumed
+ * that the compression is not enabled and no first prefix byte is present in the blobs.
+ */
+export class DocumentStorageServiceCompressionAdapter extends DocumentStorageServiceProxy {
+	private _isCompressionEnabled: boolean = false;
+	public static readonly compressionMarkupBlob = ".summary-blob-compression.enabled";
```

Comment:
@milanro - This is the "key" I mentioned last time we chatted that could be generalized.  My original idea was to make this a summary format version.  Vlad suggested something more flexible than a version number.

Reply:
@milanro - based on your feedback regarding reading the key values, I'll suggest "blobHeaders" as the name.

When the "blobHeaders" key exists, blobs beginning with 'bx' are interpreted as having a binary metadata header as implemented by this adapter.

When this key exists the corresponding value must be '1' even though the corresponding value is not used.

Example:
```
".metadata": {
    "createContainerRuntimeVersion":"0.59.1000",
    "createContainerTimestamp":1651351060440,
    "summaryFormatVersion":1,
    "gcFeature":0,
    "blobHeaders":1,          // <-- Blobs in this summary tree use metadata
}
```

Reply:
@DLehenbauer But .metadata is the key to the ISummaryBlob. The underlying key/value pairs are entries inside the blob. This would suppose to download the .metadata blob and parse it to obtain the blobHeaders value. Didn't we want to put something like .blobHeaders beside the .metadata key pointing to empty blob? This is the present way.

Reply:
Thanks for explaining.  In that case, I'd recommend '.metadata.blobHeaders':

In a follow up change, we should add the key to:
packages/runtime/container-runtime/src/summary/summaryFormat.ts

Reply:
@DLehenbauer packages/runtime/container-runtime/src/summary/summaryFormat.ts is not available in @fluidframework/driver-utils and cannot be imported as it is located in @fluidframework/container-runtime, which would cause the the circular import. The '.metadata.blobHeaders' definition is therefore in @fluidframework/driver-utils but it is mirrored and exported in packages/runtime/container-runtime/src/summary/summaryFormat.ts